index.html

```diff
@@ -2970,6 +2970,21 @@ <h2 id="http-baseline-profile-security">Security</h3>
         WoT Security Best Practices</a> for implementation advice.
       </p>
     </section>
+
+    <!-- Discovery -->
+    <section>
+      <h2 id="http-baseline-profile-discovery">Discovery</h2>
+      <p class="rfc2119-assertion" id="http-baseline-profile-discovery-1">
+        In order to conform with the HTTP Baseline Profile, a Web Thing's Thing 
```

Comment:
I think this language is problematic, since it mandates the presence of a TD server as part of a deployment of profile compliant devices and I don't understand the rationale behind it.

How does the manufacturer of a thing ensure that the TD is deployed in this way? What if all TDs are stored on an old-fashioned FTP server or are stored in the file system (or hard coded) in the consumer?

I don't see a clear business model for a 3rd party to provide a TD server with a revenue stream that pays for running the server.
I'm afraid that if we enforce that constraint, we might risk adoption of the profile.

Reply:
I sort of agree with this comment but we don´t have to make a one size fits all profile anyways, just an implementable subset is fine. If we are trying to make something that fits all, lots of other parts of the spec are open for change.

Reply:
@mlagally wrote:
> I think this language is problematic, since it mandates the presence of a TD server as part of a deployment of profile compliant devices and I don't understand the rationale behind it.

A TD server is just an HTTP server which can serve a single JSON Thing Description. A Web Thing conforming to an HTTP already has to have an HTTP server serving JSON resources, so there's no reason the Thing Description (the most important resource of all) couldn't be served over HTTP too.

> How does the manufacturer of a thing ensure that the TD is deployed in this way?

The obvious way is for the Web Thing to serve its own Thing Description, as described above. But they could host it elsewhere if they really wanted to... I don't think I understand the problem.

> What if all TDs are stored on an old-fashioned FTP server or are stored in the file system (or hard coded) in the consumer?

Then it's not compliant with the profile. I don't know why on earth a manufacturer of a greenfield HTTP-based device would want to do either of those things. Hard-coding TDs in a Consumer is particularly counterproductive as far as interoperability is concerned.

> I don't see a clear business model for a 3rd party to provide a TD server with a revenue stream that pays for running the server.

They don't need to. The Web Thing can serve the TD itself.

> I'm afraid that if we enforce that constraint, we might risk adoption of the profile.

I'm afraid that if Thing Descriptions don't have even one single basic mechanism for discovering them, it will risk adoption of the profile. Do you have another proposed mechanism other than the HTTP profile requiring the TD be served over HTTP?

I'm really surprised that having the HTTP Profile require that Thing Descriptions are served over HTTP is in any way controversial. I thought we already agreed that in the last meeting:

> Arch call on July 20th:
> It seems to ok to require that TDs have a HTTP URL, wording of the spec needs to be aligned with the discovery spec.
> Proposed way forward: upate language, resolve merge conflict.